include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -550,6 +550,8 @@ class permutation_iterator
     typedef ::std::random_access_iterator_tag iterator_category;
```

Comment:
we also need to read the `iterator_category` from `std::iterator_traits<Iterator>::iterator_category` to support more than just `random_access_iterator`

Reply:
Discussed offline. Seems `permutation_iterator` is naturally mapped on random access one. Other iterators require additional specializations and they would be inefficient in generic case. Without dedicated request there is no point to support something but `random_access_iterator`.